src/block/block_cache.c

```diff
@@ -312,24 +334,22 @@ __blkcache_estimate_filesize(WT_SESSION_IMPL *session)
 }
 
 /*
- * __wt_blkcache_get_or_check --
- *     Get a block from the cache or check if one exists.
+ * __wt_blkcache_get --
+ *     Get a block from the cache.
  */
 int
-__wt_blkcache_get_or_check(
-  WT_SESSION_IMPL *session, wt_off_t offset, size_t size, uint32_t checksum, void *data_ptr)
+__wt_blkcache_get(WT_SESSION_IMPL *session, WT_ITEM *data, const uint8_t *addr, size_t addr_size,
```

Comment:
@sulabhM, there are a couple of reasons that I didn't do it that way -- one is just in my head, and that's the feeling that `WT_NOTFOUND` is an operational error returned to the application rather than an internal state of a WT library operation. The second is sheer paranoia: imagine a future where `__wt_blkcache_get()` is more complicated and calls more underlying WT functions, and one of them returns `WT_NOTFOUND` on a real error, and we'd treat that as "OK" and keep going, when it might not be OK.

I'd be the first to admit that neither of those are "good" reasons, but they make me lean toward doing it the way it is and keeping "not-found" out of the error path.